src/builtin_string.cpp

```diff
@@ -458,39 +456,49 @@ class pcre2_matcher_t: public string_matcher_t
         // Return values: -1 = error, 0 = no match, 1 = match
         if (pcre2_rc == PCRE2_ERROR_NOMATCH)
         {
-            return 0;
+            if (opts.invert_match && !opts.quiet)
```

Comment:
This might be why `string match -rvq` seems to always return 1 - without non-matches, with just non-matches and with one non-match and a match.